### ISSUE TITLE
[Feature Request] Return current speed in test mode

### ISSUE THREAD

**Issue by Sodamin** [tum-autonomousdriving/autonomous-driving-simulator#11]
Seems like we get steering angle and throttle back from the simulation.

I would propose to rather send back at least the current speed. To allow to control for specific speeds. (We should probably also think about removing the speed limit within the simulation in test mode)

**Comment by Einholz-Gao** [comment]
Yeah, as you can see I add a speed limit module because the speed of this race car are too high without a limit. I don't think we should remove it in test mode, before we change a normal, slower car...
You can try removing the speed limit and have a look, it will run fxxx so fast...

**Comment by tarikisildar** [comment]
I think he is asking about returning the current speed information as a data like steering angle and throttle. I agree it's a must have.

**Comment by Einholz-Gao** [comment]
Hi.
Right, so do you want to send the speed back to python or just display it on the screen?
Now the speed control depends only on the throttle and brake. In Unity we have a speed limiter, too.

**Comment by Sodamin** [comment]
I think what would make sense is that the simulator gives it back, just like the image. 
Because what a car should do is also to regulate the speed by using the throttle to accellerate until the desired speed (which might be the speed limit, but maybe also below when going around corners etc.) So it is essential to have a closed speed regulation loop.

**Comment by Einholz-Gao** [comment]
Next version we can get the velocity from SocketIO in python

**Comment by BaranEkin** [comment]
Are we getting this feature soon? I also agree it is very fundamental and absolutely required.

**Comment by Einholz-Gao** [comment]
After Testing this velocity value can be sent through python code. Please after the version 0.24 updating check the py code and try it on the new simulator.

**Comment by tarikisildar** [comment]
Just to clear any possible missunderstanding, the request is the simulator sending the velocity information out (to the python code) during the runtime. simulator itself doesn't require any speed input. Training data also doesn't (this maybe can be used but probably not).

Also the need for the speed-limit within the simulator comes from the car accelerating too fast in my opinion. If we can tweak the engine torque and gear settings it'll become more managable.
